pkg/cdi/container-edits.go

```diff
@@ -138,6 +175,27 @@ func (e *ContainerEdits) Validate() error {
 	return nil
 }
 
+// Append other edits into this one. If called with a nil receiver,
+// allocates and returns newly allocated edits.
+func (e *ContainerEdits) Append(o *ContainerEdits) *ContainerEdits {
+	if o == nil || o.ContainerEdits == nil {
+		return e
+	}
+	if e == nil {
+		e = &ContainerEdits{}
+	}
+	if e.ContainerEdits == nil {
+		e.ContainerEdits = &specs.ContainerEdits{}
+	}
+
+	e.Env = append(e.Env, o.Env...)
+	e.DeviceNodes = append(e.DeviceNodes, o.DeviceNodes...)
+	e.Hooks = append(e.Hooks, o.Hooks...)
+	e.Mounts = append(e.Mounts, o.Mounts...)
```

Comment:
Question: Should this care about duplication?

Reply:
Well, this also was a conscious choice, mostly due to where/how we use it (we append during injection, then update the OCI Spec and then the whole thing is forgotten about and garbage collected) and since we don't mutate these in place once they are loaded from a Spec file. Still, it is a valid question if this is the right choice.

Reply:
For clarification, I'm OK with not solving this now.

Reply:
Okay. Let's sit on this a bit and if it feels like (deep-)copying (device nodes) would be a better choice, we can update that with a separate PR.